server/core/src/main/scala/sttp/tapir/server/interceptor/log/ServerLog.scala

```diff
@@ -93,8 +98,8 @@ case class DefaultServerLog[F[_]](
     else noLog
 
   override def decodeFailureHandled(ctx: DecodeFailureContext, response: ServerResponse[_], token: TOKEN): F[Unit] =
-    if (logWhenHandled)
-      doLogWhenHandled(
+    if (logAllDecodeFailures)
+      doLogAllDecodeFailures(
```

Comment:
I think from a use-case point of view, you might want to have a log when tapir produces a response either because the request was handled by an endpoint (when the inputs decode successfully), or when there was a decode failure, which was translated into a response. So having a single flag controlling these two situations makes sense.

You might *additionally* want to also log the decode failures which where not handled (and for which the next endpoint will be attempted). But that's a more rare case, so I'd have a separate flag here -however, with a better name.